neva

```diff
@@ -103,37 +138,42 @@ end
 local function check_cmd(cmd, msg)
     local c = string.format("command -v %s &> /dev/null", cmd)
     if not os.execute(c) then
-        print(msg)
-        os.exit(1)
+        print("[ERR] " .. msg)
+        os.exit(2)
     end
 end
 
 -- check whether requirements are ready, e.g. curl
 local function check_requirements()
-    check_cmd("curl", "require curl installed")
-    check_cmd("tar", "require tar installed")
-    check_cmd("readlink", "require readlink installed")
-    check_cmd("gojq", "require gojq installed")
+    check_cmd("curl", "Need 'curl' (command not found)")
+    check_cmd("tar", "Need 'tar' (command not found)")
+    check_cmd("readlink", "Need 'readline' (command not found)")
+    check_cmd("gojq", "Need 'gojq' (command not found)")
 
     -- create default folder
     local cmd_create = string.format("mkdir -p %s %s",
         NEVA_DL_HOME, NEVA_VER_HOME)
     if not os.execute(cmd_create) then
-        print("can't initialize neva home")
+        print("[ERR] Can't initialize neva home")
         os.exit(2)
     end
 end
 
 local NVIM_BASE_URL = "https://github.com/neovim/neovim/releases/download"
+local OS_TYPE = io.popen("uname -s", "r"):read("*l"):lower()
 
 -- url format
--- https://github.com/neovim/neovim/releases/download/nightly/nvim-macos.tar.gz
+-- https://github.com/neovim/neovim/releases/download/nightly/nvim-<OS_TYPE>.tar.gz
 local function download_url(version)
-    if version ~= "nightly" then
+    if version ~= "nightly" and version ~= "stable" then
         version = "v" .. version
     end
 
-    return NVIM_BASE_URL .. string.format("/%s/nvim-macos.tar.gz", version)
+    if OS_TYPE == "linux" then
+        return NVIM_BASE_URL .. string.format("/%s/nvim-linux64.tar.gz", version)
+    elseif OS_TYPE == "darwin" or os_type == "mac" then
```

Comment:
typo: os_type -> OS_TYPE?

Reply:
Small mistake, I forgot to capitalize it :sweat_smile: